Failed to compile hipSYCL with the latest LLVM

**Bug summary**
The following errors prevent compiling hipSYCL with LLVM which is built from its latest main branch.
```
[31/55] Building CXX object src/compiler/CMakeFiles/hipSYCL_clang.dir/HipsyclClangPlugin.cpp.obj
FAILED: src/compiler/CMakeFiles/hipSYCL_clang.dir/HipsyclClangPlugin.cpp.obj 
D:\a\1\s\llvm\build\install\bin\clang++.exe -DHIPSYCL_COMPILER_COMPONENT -DHIPSYCL_DEBUG_LEVEL=2 -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -DUNICODE -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DhipSYCL_clang_EXPORTS -ID:/a/1/s/hipSYCL/src/compiler/../../include -ID:/a/1/s/llvm/llvm/include -ID:/a/1/s/llvm/build/include -ID:/a/1/s/hipSYCL/src/compiler -ID:/a/1/s/hipSYCL/build/include -ID:/a/1/s/llvm/build/install/include -ID:/a/1/s/llvm/clang/include -ID:/a/1/s/llvm/build/tools/clang/include -O3 -DNDEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrt -fno-exceptions -std=c++17 -MD -MT src/compiler/CMakeFiles/hipSYCL_clang.dir/HipsyclClangPlugin.cpp.obj -MF src\compiler\CMakeFiles\hipSYCL_clang.dir\HipsyclClangPlugin.cpp.obj.d -o src/compiler/CMakeFiles/hipSYCL_clang.dir/HipsyclClangPlugin.cpp.obj -c D:/a/1/s/hipSYCL/src/compiler/HipsyclClangPlugin.cpp
D:/a/1/s/hipSYCL/src/compiler/HipsyclClangPlugin.cpp:56:14: error: no type named 'RegisterStandardPasses' in namespace 'llvm'
static llvm::RegisterStandardPasses
       ~~~~~~^
D:/a/1/s/hipSYCL/src/compiler/HipsyclClangPlugin.cpp:57:67: error: no member named 'EP_EnabledOnOptLevel0' in 'llvm::PassManagerBuilder'
    RegisterGlobalsPruningPassOptLevel0(llvm::PassManagerBuilder::EP_EnabledOnOptLevel0,
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~^
D:/a/1/s/hipSYCL/src/compiler/HipsyclClangPlugin.cpp:60:14: error: no type named 'RegisterStandardPasses' in namespace 'llvm'
static llvm::RegisterStandardPasses
       ~~~~~~^
D:/a/1/s/hipSYCL/src/compiler/HipsyclClangPlugin.cpp:61:71: error: no member named 'EP_OptimizerLast' in 'llvm::PassManagerBuilder'
    RegisterGlobalsPruningPassOptimizerLast(llvm::PassManagerBuilder::EP_OptimizerLast,
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~^
4 errors generated.
```

**To Reproduce**
1. Build LLVM from its latest main branch.
2. Use the built LLVM to compile hipSYCL.

**Expected behavior**
hipSYCL should be compiled properly as before.

**Describe your setup**
I think it is caused by this commit: https://github.com/llvm/llvm-project/commit/7ae6838defb21737963b1dd8ff9de7e87052c74f. It seems the class and enum used by hipSYCL were removed from LLVM recently. I'm wondering if there is a way to solve this issue. Thanks.

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,5 @@
 # CMake out-of-source
-build/
+build*/
 *-build-*/
 
 # vscode in-source configs

--- a/src/compiler/HipsyclClangPlugin.cpp
+++ b/src/compiler/HipsyclClangPlugin.cpp
@@ -48,6 +48,7 @@ namespace compiler {
 static clang::FrontendPluginRegistry::Add<hipsycl::compiler::FrontendASTAction>
     HipsyclFrontendPlugin{"hipsycl_frontend", "enable hipSYCL frontend action"};
 
+#if LLVM_VERSION_MAJOR < 16
 static void registerGlobalsPruningPass(const llvm::PassManagerBuilder &,
                                        llvm::legacy::PassManagerBase &PM) {
   PM.add(new GlobalsPruningPassLegacy{});
@@ -84,7 +85,9 @@ static void registerMarkParallelPass(const llvm::PassManagerBuilder &,
 static llvm::RegisterStandardPasses
     RegisterMarkParallelBeforeVectorizer(llvm::PassManagerBuilder::EP_VectorizerStart,
                                          registerMarkParallelPass);
-#endif
+#endif // HIPSYCL_WITH_ACCELERATED_CPU
+#endif // LLVM_VERSION_MAJOR < 16
+
 #if !defined(_WIN32) && LLVM_VERSION_MAJOR >= 11
 #define HIPSYCL_RESOLVE_AND_QUOTE(V) #V
 #define HIPSYCL_STRINGIFY(V) HIPSYCL_RESOLVE_AND_QUOTE(V)

--- a/src/compiler/cbs/PipelineBuilder.cpp
+++ b/src/compiler/cbs/PipelineBuilder.cpp
@@ -47,6 +47,7 @@
 
 namespace hipsycl::compiler {
 
+#if LLVM_VERSION_MAJOR < 16
 void registerCBSPipelineLegacy(llvm::legacy::PassManagerBase &PM) {
   HIPSYCL_DEBUG_WARNING << "CBS pipeline might not result in peak performance with old PM\n";
   PM.add(new LoopSplitterInliningPassLegacy{});
@@ -67,6 +68,7 @@ void registerCBSPipelineLegacy(llvm::legacy::PassManagerBase &PM) {
   PM.add(new KernelFlatteningPassLegacy{});
   PM.add(new LoopsParallelMarkerPassLegacy{});
 }
+#endif // LLVM_VERSION_MAJOR < 16
 
 void registerCBSPipeline(llvm::ModulePassManager &MPM, OptLevel Opt) {
   MPM.addPass(SplitterAnnotationAnalysisCacher{});
@@ -83,8 +85,10 @@ void registerCBSPipeline(llvm::ModulePassManager &MPM, OptLevel Opt) {
     FPM.addPass(llvm::InstCombinePass{});
 #if LLVM_VERSION_MAJOR <= 13
     FPM.addPass(llvm::SROA{});
-#else
+#elif LLVM_VERSION_MAJOR < 16
     FPM.addPass(llvm::SROAPass{});
+#else
+    FPM.addPass(llvm::SROAPass{llvm::SROAOptions::ModifyCFG});
 #endif
 
     FPM.addPass(llvm::SimplifyCFGPass{});

--- a/src/compiler/cbs/SubCfgFormation.cpp
+++ b/src/compiler/cbs/SubCfgFormation.cpp
@@ -84,7 +84,7 @@ getLocalSizeArgumentFromAnnotation(llvm::Function &F) {
   for (auto &BB : F)
     for (auto &I : BB)
       if (auto *UI = llvm::dyn_cast<llvm::CallInst>(&I))
-        if (UI->getCalledFunction()->getName().equals("llvm.var.annotation")) {
+        if (UI->getCalledFunction()->getName().startswith("llvm.var.annotation")) {
           HIPSYCL_DEBUG_INFO << *UI << '\n';
           llvm::GlobalVariable *AnnotateStr = nullptr;
           if (auto *CE = llvm::dyn_cast<llvm::ConstantExpr>(UI->getOperand(1));

--- a/tests/compiler/lit.cfg.py
+++ b/tests/compiler/lit.cfg.py
@@ -13,4 +13,5 @@ config.substitutions.append(('%syclcc', config.hipsycl_syclcc))
 
 if "HIPSYCL_DEBUG_LEVEL" in os.environ:
   config.environment["HIPSYCL_DEBUG_LEVEL"] = os.environ["HIPSYCL_DEBUG_LEVEL"]
-  
+if "HIPSYCL_VISIBILITY_MASK" in os.environ:
+  config.environment["HIPSYCL_VISIBILITY_MASK"] = os.environ["HIPSYCL_VISIBILITY_MASK"]

